django/db/backends/mysql/operations.py

```diff
@@ -27,6 +27,8 @@ def date_extract_sql(self, lookup_type, field_name):
             # other database backends.
             # Mode 3: Monday, 1-53, with 4 or more days this year.
             return "WEEK(%s, 3)" % field_name
+        elif lookup_type == 'week_day_index':
+            return "WEEKDAY(%s)" % field_name
```

Comment:
Is the only difference with `__week_day` (see lines 21-24 above) that this one is 0-indexed instead of 1-indexed?